src/api/dependencies.py

```
@@ -0,0 +1,631 @@
+            def __init__(self, router):
+                self.router = router
+            
+            async def process_file(self, file_path: str) -> ProcessingResult:
+                """Process file using embedding pipeline."""
+                try:
+                    # Use the embedding pipeline for complete processing
+                    pipeline = get_retrieval_system()
+                    
+                    # Process document through the pipeline: chunk → embed → store
+                    result = pipeline.process_document(file_path)
+                    return result
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)